ui/src/store.js

```diff
@@ -0,0 +1,186 @@
+import autodux from 'autodux';
+import { doFetch } from './utils/fetch-websocket';
+
+export const {
+  reducer,
+  initial: defaultState,
+  actions: {
+    activateConnection,
+    deactivateConnection,
+    serverConnected,
+    serverDisconnected,
+    updatePurses,
+    changePurse,
+    swapInputs,
+    changeAmount,
+    createOffer,
+    resetState,
+    updateInvitationDepositId,
+  },
+} = autodux({
+  slice: 'autoswap',
+  initial: {
+    active: false,
+    connected: false,
+    account: null,
+    purses: null,
+    inputPurse: null,
+    outputPurse: null,
+    inputAmount: null,
+    outputAmount: null,
+    freeVariable: null,
+    invitationDepositId: null,
+  },
+  actions: {
+    activateConnection: state => ({ ...state, active: true }),
+    deactivateConnection: state => ({ ...state, active: false }),
+    serverConnected: state => ({ ...state, connected: true }),
+    serverDisconnected: state => ({ ...state, connected: false }),
+    updatePurses: (state, purses) => ({ ...state, purses }),
+    changePurse: {
+      // Map positional arguments.
+      create: (purse, fieldNumber, freeVariable = null) => ({
+        purse,
+        fieldNumber,
+        freeVariable,
+      }),
+      reducer: (state, { purse, fieldNumber, freeVariable }) => {
```

Comment:
this desperately needs comments. I have no idea what this is for

Reply:
I've tried hard to "first do no harm".  This introduction of autodux doesn't improve anything except the boilerplate around the implementation of the store.  I don't fully understand the intent or use of `fieldNumber` and `freeVariable`, and I'm pretty sure there are bugs in it.

Marking with ⛵ for "this ship has sailed."